src/Credentials/CredentialProvider.php

```diff
@@ -102,7 +102,13 @@ public static function defaultProvider(array $config = [])
             );
         }
 
-        if (!empty(getenv(EcsCredentialProvider::ENV_URI))) {
+        $shouldUseEcsCredentialsProvider = getenv(EcsCredentialProvider::ENV_URI);
+        // getenv() is not thread safe - fall back to $_SERVER
+        if ($shouldUseEcsCredentialsProvider === false) {
+            $shouldUseEcsCredentialsProvider = isset($_SERVER[EcsCredentialProvider::ENV_URI]) ? $_SERVER[EcsCredentialProvider::ENV_URI] : false;
```

Comment:
Depending on who is running this and on what platform $_ENV may be worth checking too- that can be a later addition though

Reply:
> In both the CGI and FastCGI SAPIs, $_SERVER is also populated by values from the environment;
https://www.php.net/manual/en/ini.core.php#ini.variables-order

I think using `$_SERVER` should be fine as a fallback. 
Is it possible that `getenv()` and `$_SERVER` return nothing but `$_ENV`contains values? 
The only example I can think of is  explicitly set with `$_ENV['foo'] = 'bar'`, but that's a rear edge case.

Reply:
agreed, and thank you for getting those fixes in so quickly as well as for your contribution :) I will merge soon